### PR TITLE
[AT-3197] Adding deprecation message VIZ features

### PR DESCRIPTION
**Pull request by andher1802**
Adding deprecation warnings to the visualization features in Job, JobCollection.

Items:
* [ ] Ran test & live-tests
* [ ] Implemented (new) unit tests
* [ ] Removed credentials
* [ ] Removed argument pointing to staging
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [ ] Bumped version
* [ ] Added changelog
* [ ] Updated announcement banner

### CODE REVIEW

**Comment by dustgalactic** [comment]
@andher1802 Should I add a deprecation banner to instances of these functions throughout the reference + remove the https://sdk.up42.com/visualizations/ page?

File: up42/viztools.py [hunk 1]
```diff
@@ -137,6 +138,12 @@ def plot_results(
                  e.g. matplotlib cmap etc.
         """
         warnings.filterwarnings("ignore", category=rasterio.errors.NotGeoreferencedWarning)
+        warn(
+            "Visualization methods are getting deprecated. The current feature will be discontinued "
+            "after March 31, 2024, and will be replaced by new processing functionalities.",
```
Comment (seedlit) [inline up42/viztools.py:143]: If I am not wrong, we aren't providing any alternative for visualization in new processing functionalities right?
In this case, consider removing "and will be replaced by new processing functionalities."

Comment (seedlit) [reply up42/viztools.py:143]: same for others below.

**Review by seedlit — COMMENTED** [review]
left a comment :)

File: up42/viztools.py (hunk 1 above)
Comment (andher1802) [reply up42/viztools.py:143]: I will add your suggestion thanks. I think adding new functionalities can be redundant.

But please be aware that this deprecation message will be exposed when the user runs this function from Job, JobCollection, and JobTask classes. I was referring to these classes when I added the new functionalities. Just to make sure we are all aware of the context.

Comment (seedlit) [reply up42/viztools.py]: I see.
Maybe we can control that by updating the `stacklevel=2` param? 🤔

**Comment by dustgalactic** [comment]
@andher1802 I've removed the page about visualization and mentions of the functionality throughout the documentation. As for functions, I've added a "This function is deprecated." banner. IMHO, this is different from the discontinuation announcement we have + the Catalog class also uses some of these functions.

File: up42/viztools.py [hunk 2]
```diff
@@ -137,6 +138,12 @@ def plot_results(
                  e.g. matplotlib cmap etc.
         """
         warnings.filterwarnings("ignore", category=rasterio.errors.NotGeoreferencedWarning)
+        warn(
+            "Visualization methods are getting deprecated. The current feature will be discontinued "
```
Comment (dustgalactic) [inline up42/viztools.py:142]: Visualization methods are getting deprecated -> Visualization methods are deprecated

**Review by seedlit — APPROVED** [review]
LGTM!
Just one question: stacklevel=2 means the error will be surfaced in the parent function right?

**Pull request merged**